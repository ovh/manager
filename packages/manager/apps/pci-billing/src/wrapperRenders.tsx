import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR', currency: 'â‚¬' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
