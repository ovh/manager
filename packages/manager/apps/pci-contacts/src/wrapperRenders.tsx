import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        return Promise.resolve('');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
