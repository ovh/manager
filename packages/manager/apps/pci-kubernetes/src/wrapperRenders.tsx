import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';

export const shellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: () => ({
      ovhSubsidiary: 'mocked_ovhSubsidiary',
      currency: { code: 'EUR' },
    }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
