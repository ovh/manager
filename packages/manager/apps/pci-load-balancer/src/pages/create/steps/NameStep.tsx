import {
  OsdsButton,
  OsdsFormField,
  OsdsInput,
  OsdsText,
} from '@ovhcloud/ods-components/react';
import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
import {
  ODS_BUTTON_VARIANT,
  ODS_INPUT_TYPE,
  ODS_TEXT_LEVEL,
  ODS_TEXT_SIZE,
} from '@ovhcloud/ods-components';
import {
  StepComponent,
  useNotifications,
} from '@ovh-ux/manager-react-components';
import { Translation, useTranslation } from 'react-i18next';
import { ApiError } from '@ovh-ux/manager-core-api';
import { useNavigate, useParams } from 'react-router-dom';
import { useTracking } from '@ovh-ux/manager-react-shell-client';
import { useState } from 'react';
import {
  LOAD_BALANCER_CREATION_TRACKING,
  LOAD_BALANCER_NAME_REGEX,
} from '@/constants';
import { StepsEnum, useCreateStore } from '@/pages/create/store';
import { useGetFlavor } from '@/api/hook/useFlavors';
import queryClient from '@/queryClient';
import { getAllLoadBalancersQueryKey } from '@/api/hook/useLoadBalancer';

export const NameStep = (): JSX.Element => {
  const { t: tCommon } = useTranslation('pci-common');
  const { t: tCreate } = useTranslation('load-balancer/create');

  const [didTryCreate, setDidTryCreate] = useState(false);

  const { projectId } = useParams();
  const { trackClick, trackPage } = useTracking();

  const store = useCreateStore();

  const { data: flavor } = useGetFlavor(
    projectId,
    store.region?.name,
    store.addon,
  );

  const navigate = useNavigate();
  const { addError, addInfo } = useNotifications();

  const create = async () => {
    setDidTryCreate(true);

    trackClick({
      name: LOAD_BALANCER_CREATION_TRACKING.SUBMIT,
      type: 'action',
    });

    trackClick({
      name: `${LOAD_BALANCER_CREATION_TRACKING.CONFIRM}::${store.addon.code}::${store.region.name}`,
      type: 'action',
    });

    await store.create(
      flavor,
      () => {
        trackPage({
          name: LOAD_BALANCER_CREATION_TRACKING.SUCCESS,
          type: 'navigation',
        });
        addInfo(
          <Translation ns="load-balancer/create">
            {(_t) => _t('octavia_load_balancer_create_banner')}
          </Translation>,
          false,
        );
        navigate('..');
        queryClient.invalidateQueries({
          queryKey: getAllLoadBalancersQueryKey(projectId),
        });
      },
      (error: ApiError) => {
        trackPage({
          name: LOAD_BALANCER_CREATION_TRACKING.ERROR,
          type: 'navigation',
        });
        addError(
          <Translation ns="load-balancer">
            {(_t) => (
              <span
                dangerouslySetInnerHTML={{
                  __html: _t('octavia_load_balancer_global_error', {
                    message: ((error.response as unknown) as {
                      data: { message: string };
                    }).data.message,
                    requestId: ((error.response as unknown) as {
                      headers: Record<string, unknown>;
                    }).headers['x-ovh-queryid'],
                  }),
                }}
              ></span>
            )}
          </Translation>,
          false,
        );
      },
    );
  };

  const cancel = () => {
    trackClick({
      name: LOAD_BALANCER_CREATION_TRACKING.CANCEL,
      type: 'action',
    });
    store.reset();
    navigate('..');
  };

  return (
    <StepComponent
      title={tCreate('octavia_load_balancer_create_name_field_label')}
      isOpen={store.steps.get(StepsEnum.NAME).isOpen}
      isChecked={store.steps.get(StepsEnum.NAME).isChecked}
      isLocked={store.steps.get(StepsEnum.NAME).isLocked}
      order={6}
    >
      <OsdsFormField
        className="mt-8 w-[20rem]"
        inline
        error={(() => {
          if (store.name.length > 70)
            return tCommon('common_field_error_maxlength', {
              maxlength: 70,
            });
          if (store.name.length && !store.name.match(LOAD_BALANCER_NAME_REGEX))
            return tCommon('common_field_error_pattern');

          return '';
        })()}
      >
        <OsdsText
          color={ODS_THEME_COLOR_INTENT.text}
          size={ODS_TEXT_SIZE._100}
          level={ODS_TEXT_LEVEL.subheading}
          slot="label"
        >
          {tCreate('octavia_load_balancer_create_name_field_label')}
        </OsdsText>
        <OsdsText
          color={ODS_THEME_COLOR_INTENT.text}
          size={ODS_TEXT_SIZE._100}
          level={ODS_TEXT_LEVEL.subheading}
          slot="helper"
        >
          {tCreate('octavia_load_balancer_create_name_field_help')}
        </OsdsText>
        <OsdsInput
          value={store.name}
          type={ODS_INPUT_TYPE.text}
          onOdsValueChange={(event) =>
            store.set.name(event.target.value as string)
          }
          className={
            (store.name.length &&
              !store.name.match(LOAD_BALANCER_NAME_REGEX)) ||
            store.name.length > 70
              ? 'bg-red-100 border-red-500 text-red-500 focus:text-red-500'
              : 'border-color-[var(--ods-color-default-200)] bg-white'
          }
        />
      </OsdsFormField>
      {!didTryCreate && (
        <div className="mt-8">
          <OsdsButton
            inline
            color={ODS_THEME_COLOR_INTENT.info}
            onClick={create}
            {...((store.name.length &&
              !store.name.match(LOAD_BALANCER_NAME_REGEX)) ||
            store.name.length > 70
              ? { disabled: true }
              : {})}
          >
            {tCreate('octavia_load_balancer_create_title')}
          </OsdsButton>
          <OsdsButton
            inline
            variant={ODS_BUTTON_VARIANT.ghost}
            color={ODS_THEME_COLOR_INTENT.info}
            onClick={cancel}
          >
            {tCommon('common_cancel')}
          </OsdsButton>
        </div>
      )}
    </StepComponent>
  );
};
