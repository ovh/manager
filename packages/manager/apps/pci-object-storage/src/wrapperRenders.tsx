import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import React from 'react';

export const shellContext = {
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

export const wrapper = ({ children }) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider
      value={(shellContext as unknown) as ShellContextType}
    >
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);

export const mockShellContext = {
  environment: {
    getRegion: vi.fn().mockReturnValue('EU'),
    getUser: vi.fn().mockReturnValue({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
    getApplication: vi.fn(),
  },
  shell: {
    environment: {
      getRegion: vi.fn().mockReturnValue('EU'),
      getUser: vi.fn().mockReturnValue({}),
      getApplication: vi.fn(),
    },
    navigation: {
      getURL: vi.fn(),
    },
    tracking: {
      trackPage: vi.fn(),
      trackClick: vi.fn(),
    },
  },
};

export const wrapperShow = ({ children }: { children: React.ReactNode }) => {
  const router = createMemoryRouter(
    [
      {
        path: '/',
        element: children,
        handle: {
          crumb: () => 'Home',
        },
      },
    ],
    {
      initialEntries: ['/'],
      future: {
        v7_startTransition: true,
        v7_relativeSplatPath: true,
      } as any,
    },
  );

  return (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={mockShellContext as any}>
        <RouterProvider router={router} />
      </ShellContext.Provider>
    </QueryClientProvider>
  );
};

export const wrapperOffsiteReplication = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <QueryClientProvider client={queryClient}>
    <ShellContext.Provider value={mockShellContext as any}>
      {children}
    </ShellContext.Provider>
  </QueryClientProvider>
);
