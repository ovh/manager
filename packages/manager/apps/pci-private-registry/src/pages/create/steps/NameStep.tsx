import {
  OsdsFormField,
  OsdsInput,
  OsdsText,
} from '@ovhcloud/ods-components/react';
import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
import { ODS_INPUT_TYPE, ODS_TEXT_SIZE } from '@ovhcloud/ods-components';
import { StepComponent } from '@ovh-ux/manager-react-components';
import { useTranslation } from 'react-i18next';
import { StepEnum } from '@/pages/create/types';
import { useStore } from '@/pages/create/store';

export default function NameStep() {
  const { t: tCreate } = useTranslation('create');
  const { t: tCommon } = useTranslation('common');
  const { t: tCommonField } = useTranslation('common_field');

  const store = useStore();

  const { stepsHandle } = store;

  return (
    <StepComponent
      isOpen={store.stepsState[StepEnum.NAME].isOpen}
      isLocked={store.stepsState[StepEnum.NAME].isLocked}
      isChecked={store.stepsState[StepEnum.NAME].isChecked}
      order={2}
      title={tCreate('private_registry_create_name_cluster')}
      next={{
        action: () => {
          stepsHandle.check(StepEnum.NAME);
          stepsHandle.lock(StepEnum.NAME);

          stepsHandle.open(StepEnum.PLAN);
        },
        label: tCommonField('common_stepper_next_button_label'),
        isDisabled: !store.state.name.value.trim().length,
      }}
      edit={{
        action: () => {
          stepsHandle.close(StepEnum.PLAN);
          stepsHandle.uncheck(StepEnum.PLAN);
          stepsHandle.unlock(StepEnum.PLAN);

          stepsHandle.uncheck(StepEnum.NAME);
          stepsHandle.unlock(StepEnum.NAME);
        },
        label: tCommonField('common_stepper_modify_this_step'),
      }}
    >
      <OsdsFormField
        className="mt-4"
        inline
        error={
          store.state.name.touched && !store.state.name.value.trim().length
            ? tCommonField('common_field_error_required')
            : ''
        }
      >
        <OsdsText
          slot="label"
          color={
            store.state.name.touched && !store.state.name.value.trim().length
              ? ODS_THEME_COLOR_INTENT.error
              : ODS_THEME_COLOR_INTENT.text
          }
          className="mt-4"
          size={ODS_TEXT_SIZE._100}
        >
          {tCommon('private_registry_create_name_label')}
        </OsdsText>
        <OsdsInput
          value={store.state.name.value}
          inline
          color={ODS_THEME_COLOR_INTENT.primary}
          onOdsValueChange={(e) => {
            store.set.name.touched(true);
            store.set.name.value(e.detail.value);
          }}
          type={ODS_INPUT_TYPE.text}
          error={store.state.name.touched && !store.state.name.value.length}
          className="border"
        />
      </OsdsFormField>
    </StepComponent>
  );
}
