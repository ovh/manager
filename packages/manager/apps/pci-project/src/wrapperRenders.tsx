import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactNode } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { vi } from 'vitest';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'FR' }),
    getRegion: () => 'EU',
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn().mockImplementation((app, path) => {
        if (app === 'dedicated' && path === '#/billing/history') {
          return Promise.resolve('https://billing-url');
        }
        if (app === 'dedicated' && path === '#/contacts/services') {
          return Promise.resolve('https://contacts-url');
        }
        return Promise.resolve('https://mocked-url');
      }),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = (
  contextValue: ShellContextType = shellContext,
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        staleTime: 300_000,
      },
    },
  });

  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={contextValue}>
        <MemoryRouter>{children}</MemoryRouter>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
