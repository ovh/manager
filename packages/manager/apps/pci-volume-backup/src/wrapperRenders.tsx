import { ReactNode } from 'react';
import { QueryClientProvider } from '@tanstack/react-query';
import {
  ShellContext,
  ShellContextType,
} from '@ovh-ux/manager-react-shell-client';
import { vi } from 'vitest';
import { ErrorBoundary } from 'react-error-boundary';
import queryClient from './queryClient';

export const shellContext = ({
  environment: {
    getUser: () => ({ ovhSubsidiary: 'mocked_ovhSubsidiary' }),
  },
  shell: {
    navigation: {
      navigateTo: vi.fn(),
      getURL: vi.fn(),
    },
  },
} as unknown) as ShellContextType;

export const createWrapper = () => {
  const Wrapper = ({ children }: { children: ReactNode }) => (
    <QueryClientProvider client={queryClient}>
      <ShellContext.Provider value={shellContext}>
        <ErrorBoundary fallbackRender={() => <div>error boundary</div>}>
          {children}
        </ErrorBoundary>
      </ShellContext.Provider>
    </QueryClientProvider>
  );
  return Wrapper;
};
