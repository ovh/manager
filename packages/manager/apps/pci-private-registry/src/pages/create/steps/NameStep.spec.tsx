import { render } from '@testing-library/react';
import { Mock, vi } from 'vitest';
import { StepComponent } from '@ovh-ux/manager-react-components';
import { wrapper } from '@/wrapperRenders';
import NameStep from '@/pages/create/steps/NameStep';

const compute = () => render(<NameStep />, { wrapper });

describe('NameStep', () => {
  vi.mock('@ovh-ux/manager-react-components', (importOriginal) => ({
    ...importOriginal,
    StepComponent: vi
      .fn()
      .mockImplementation(() => (
        <div data-testid="StepComponent">StepComponent</div>
      )),
  }));

  it('should render', () => {
    const { container } = compute();
    expect(container).toMatchSnapshot();
  });

  it('should render StepComponent', () => {
    compute();

    const spy = StepComponent as Mock;

    expect(spy).toHaveBeenCalled();
  });
});
