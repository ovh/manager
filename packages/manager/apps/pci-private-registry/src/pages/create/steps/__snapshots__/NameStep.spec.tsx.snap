// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NameStep > should render 1`] = `
<div>
  <div
    data-testid="StepComponent"
  >
    StepComponent
  </div>
</div>
`;
